llm_bioinformatics_research/src/UserProfile/UserProfile.js

```diff
@@ -0,0 +1,152 @@
+    const file = e.target.files[0];
+    if (file) {
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        setUser({ ...user, image: reader.result }); 
+      };
+      reader.readAsDataURL(file);
+    }
+  };
+
+  const handleLogout = () => {
+    navigate('/login'); // Placeholder for actual logout logic
+  };
+
+  return (
+    <Box className="userProfile-container">
+      <AppBar position="static" className="appBar">
+        <Toolbar>
+          <Typography variant="h4" className="title">
+            User Profile
+          </Typography>
+          <IconButton color="inherit" onClick={() => navigate('/home')}>
+            <CloseIcon />
+          </IconButton>
+        </Toolbar>
+      </AppBar>
+
+      <Box className="button-section">
+        <Button
+          onClick={() => setSection('profile')}
+          className={section === 'profile' ? 'selected-button' : 'default-button'}
+        >
+          <ManageAccounts /> My Profile
+        </Button>
+
+        <Button
+          onClick={() => setSection('settings')}
+          className={section === 'settings' ? 'selected-button' : 'default-button'} 
+        >
+          <Settings /> Settings
+        </Button>
+
+        <Button onClick={handleLogout} className="logout-button" aria-label='logoutbutton'>
+          <Logout /> Logout
+        </Button>
+      </Box>
+
+      {section === 'profile' && (
+        <Box className="profile-section">
+          <Box className="avatar-container">
+            <Avatar alt="User Image" src={user.image} className="avatar" />
+            <IconButton className="edit-icon" component="label">
+              <EditIcon />
+              <input type="file" hidden accept="image/*" onChange={handleImageChange} data-testid ='avatar-upload' />
+            </IconButton>
+          </Box>
+
+          <TextField
+            label="Name"
+            name="name"
+            value={user.name}
+            onChange={handleInputChange}
+            className="input-field"
+            margin="normal"
+          />
+          <TextField
+            label="Email"
+            name="email"
+            value={user.email}
+            onChange={handleInputChange}
+            className="input-field"
+            margin="normal"
+          />
+          <TextField
+            label="Phone Number"
+            name="phone"
+            value={user.phone}
+            onChange={handleInputChange}
+            className="input-field"
+            margin="normal"
+          />
+          <TextField
+            label="Location"
+            name="location"
+            value={user.location}
+            onChange={handleInputChange}
+            className="input-field"
+            margin="normal"
+          />
+        </Box>
+      )}
+
+      {section === 'settings' && (
+        <Box className="settings-section">
+          <FormControl className="input-field" margin="normal">
+            <InputLabel>Theme</InputLabel>
```

Comment:
The Theme and Language labels are positioned so that they slightly overlap with the dropdown box where the user makes these selections. They might need to be shifted up a little so that there isn't overlap. Here's what it looks like on my end:
![image](https://github.com/user-attachments/assets/d5309e08-54f5-49fd-919c-a5ed43c2ddd1)